prescient.el

```diff
@@ -119,6 +119,25 @@ be `literal+initialism', which equivalent to the list (`literal'
           (const :tag "Prefix" prefix)
           (const :tag "Anchored" anchored)))
 
+(defcustom prescient-filter-alist
+  '((literal . prescient--literal-regexp)
+    (initialism . prescient--initials-regexp)
+    (regexp . prescient--regexp-regexp)
+    (fuzzy . prescient--fuzzy-regexp)
+    (prefix . prescient--prefix-regexp)
+    (anchored . prescient--anchored-regexp))
```

Comment:
I think that if we make this alist public, we should also make the regexp generating functions public. Probably we should make some of the utility functions used in those regexp functions public as well, like `prescient--with-group`. That would be good practice to make the customization user-accessible.

Reply:
I've changed and listed the README the names of these and `prescient-with-group`. Are their other that should be included?

What does "making public" mean beyond that?

Reply:
Sorry for the delay. By "make public" I just mean rename so that they do not have a double hyphen `--`, and document in the README.